processing/src/main/java/org/apache/druid/segment/SimpleQueryableIndex.java

```diff
@@ -46,8 +47,8 @@ public class SimpleQueryableIndex implements QueryableIndex
   private final BitmapFactory bitmapFactory;
   private final Map<String, Supplier<ColumnHolder>> columns;
   private final SmooshedFileMapper fileMapper;
-  @Nullable
-  private final Metadata metadata;
+  @NotNull
+  private final Supplier<Metadata> metadataSupplier;
```

Comment:
this doesn't seem right; 
why soften `Metadata` with a `Supplier` ?

Reply:
I'm not 100% certain that this is why it was done, but one benefit of softening it could be to avoid holding the object in memory which potentially softens memory pressure.  There's probably a legitimate question to be asked if this should really be a `Supplier<>` or if `SimpleQueryableIndex` should be made an abstract class.

Reply:
Converted it to an abstract class